src/porepy/numerics/interface_laws/elliptic_interface_laws.py

```diff
@@ -851,3 +851,179 @@ def assemble_matrix_rhs(
+        # Compute effective permeability
+        k: pp.SecondOrderTensor = parameter_dictionary_h["second_order_tensor"]
+        if g_h.dim == 2:
+            R = pp.map_geometry.project_plane_matrix(g_h.nodes)
+            k = k.copy()
+            k.values = np.tensordot(R.T, np.tensordot(R, k.values, (1, 0)), (0, 1))
+            k.values = np.delete(k.values, (2), axis=0)
+            k.values = np.delete(k.values, (2), axis=1)
+        kx = k.values[0, 0]
+        ky = k.values[1, 1]
+        ke = np.sqrt(kx * ky)
+
+        # Kh is ke * specific volume, but this is already captured by k
+        Kh = proj_h * ke * mg.cell_volumes
+        inv_WI = sps.diags((np.log(proj_h * r_e / r_w) + skin_factor) / 2 * np.pi * Kh)
+        matrix_dictionary_edge[self.well_discr_matrix_key] = -inv_WI
+
+        ## Vector source.
+        # This contribution is last term of
+        # lambda = -\int{\kappa_n [p_l - p_h +  a/2 g \cdot n]} dV,
+        # where n is the outwards normal and the integral is taken over the mortar cell.
+        # (Note: This assumes a P0 discretization of mortar fluxes).
+
+        # Ambient dimension of the problem, as specified for the higher-dimensional
+        # neighbor.
+        # IMPLEMENTATION NOTE: The default value is needed to avoid that
+        # ambient_dimension becomes a required parameter. If neither ambient dimension,
+        # nor the actual vector_source is specified, there will be no problems (in the
+        # assembly, a zero vector soucre of a size that fits with the discretization is
+        # created). If a vector_source is specified, but the ambient dimension is not,
+        # a dimension mismatch will result unless the ambient dimension implied by
+        # the size of the vector source matches g_h.dim. This is okay for domains with
+        # no subdomains with co-dimension more than 1, but will fail for fracture
+        # intersections. The default value is thus the least bad option in this case.
+        vector_source_dim: int = parameter_dictionary_h.get(
+            "ambient_dimension", g_h.dim
+        )
+        # The ambient dimension cannot be less than the dimension of g_h.
+        # If this is broken, we risk ending up with zero normal vectors below, so it is
+        # better to break this off now
+        if vector_source_dim < g_h.dim:
+            raise ValueError(
+                "Ambient dimension cannot be lower than the grid dimension"
+            )
+
+        # Construct the dot product between vectors connecting fracture and well
+        # cell centers and the identity matrix.
+
+        # Project the vectors, we need to do some transposes to get this right.
+        # Note that in the codim 2 case, proj_h maps from g_h cells, not faces.
+        vectors_mortar = (proj_h * g_h.cell_centers.T - proj_l * g_l.cell_centers.T).T
+
+        # The values in vals are sorted by the mortar cell index ordering (proj is a
+        # csr matrix).
```

Comment:
Should we do a check on that? I think this is a more general comment, we make assumptions on the matrix format here and there in the code.

Reply:
Frankly not sure. Copied from RobinCoupling